testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java

```diff
@@ -241,6 +248,11 @@ else if (getHiveVersionMajor() == 3 && isFormat.test("orc")) {
             hiveValueForCaseChangeField = "\"LOWER2UPPERCASE\":2";
         }
 
+        // For ORC when we coerce NaN to String, it returns coerced value as `null`
```

Comment:
is this a bug that we should track?

Reply:
It is how ORC file behaves, I'm not sure if we need to track them in ORC project